src/common/imageio_gm.c

```diff
@@ -119,6 +119,15 @@ dt_imageio_retval_t dt_imageio_open_gm(dt_image_t *img, const char *filename, dt
     }
   }
 
+  size_t profile_length;
+  const gchar* profile_data = (gchar*) GetImageProfile(image, "ICM", &profile_length);
+  if(profile_data)
```

Comment:
For extra paranoia, I usually do `if(profile_lenght > 0 && profile_data)`

Reply:
I also like to use extra paranoia in cases where there is some uncertainty.

But in this case, the documentation clearly says that the function returns a null pointer if there is no profile:

http://www.graphicsmagick.org/api/profile.html#getimageprofile

Reply:
The documentation does not say If "no profile" is the same as "random pointer to an array of length zero"? 😉 Shouldn't happen, I know, but...

Reply:
The documentation literally says the following: "A null pointer is returned if the named profile is not present.".

In other words, `((void *)0)` is returned if the requested profile type does not exist in the image.